drivers/usb/device/usb_dc_nrfx.c

```diff
@@ -773,59 +766,81 @@ static inline void usbd_work_process_recvreq(struct nrf_usbd_ctx *ctx,
+
+static inline void usbd_work_process_ep_events(struct usbd_ep_event *ep_evt)
 {
-	struct nrf_usbd_ctx *ctx;
-	struct usbd_ep_event *ev;
+	struct nrf_usbd_ctx *ctx = get_usbd_ctx();
+	struct nrf_usbd_ep_ctx *ep_ctx = ep_evt->ep;
 
-	ctx = CONTAINER_OF(item, struct nrf_usbd_ctx, usb_work);
+	__ASSERT_NO_MSG(ep_ctx);
 
-	if (ctx->flags) {
-		if (ctx->flags & BIT(NRF_USB_STATE_CHANGE)) {
-			usbd_handle_state_change(ctx);
-			ctx->flags &= ~BIT(NRF_USB_STATE_CHANGE);
-		}
+	switch (ep_evt->evt_type) {
+	case EP_EVT_SETUP_RECV: {
+		usbd_work_process_setup(ep_ctx);
+		break;
+	}
+	case EP_EVT_RECV_REQ: {
+		usbd_work_process_recvreq(ctx, ep_ctx);
+		break;
+	}
+	case EP_EVT_RECV_COMPLETE:
+		ep_ctx->cfg.cb(ep_ctx->cfg.addr,
+			       USB_DC_EP_DATA_OUT);
+		break;
 
-		if (ctx->flags & BIT(NRF_USB_STATUS_CHANGE)) {
-			usbd_handle_status_change(ctx);
-			ctx->flags &= ~BIT(NRF_USB_STATUS_CHANGE);
+	case EP_EVT_WRITE_COMPLETE:
+		if (ep_ctx->cfg.type == USB_DC_EP_CONTROL) {
+			k_mutex_lock(&ctx->drv_lock, K_FOREVER);
+			nrfx_usbd_setup_clear();
+			k_mutex_unlock(&ctx->drv_lock);
 		}
+		ep_ctx->cfg.cb(ep_ctx->cfg.addr,
+			       USB_DC_EP_DATA_IN);
+		break;
+	default:
+		break;
 	}
+}
+
+/* Work handler */
+static void usbd_work_handler(struct k_work *item)
+{
+	struct nrf_usbd_ctx *ctx;
+	struct usbd_event *ev;
+
+	ctx = CONTAINER_OF(item, struct nrf_usbd_ctx, usb_work);
 
 	while ((ev = usbd_evt_get()) != NULL) {
-		if (!ctx->attached) {
-			LOG_ERR("USBD event dropped (not attached): %d.",
-				(uint32_t)ev->evt);
-		} else {
-			struct nrf_usbd_ep_ctx *ep_ctx = ev->ep;
 
-			switch (ev->evt) {
-			case EP_EVT_SETUP_RECV: {
-				usbd_work_process_setup(ep_ctx);
-				break;
+		switch (ev->evt_type) {
+		case USBD_EVT_EP:
+			if (!ctx->attached) {
+				LOG_ERR("EP %d event dropped (not attached).",
+					(u32_t)ev->evt.ep_evt.ep->cfg.addr);
 			}
-			case EP_EVT_RECV_REQ: {
-				usbd_work_process_recvreq(ctx, ep_ctx);
-				break;
+			usbd_work_process_ep_events(&ev->evt.ep_evt);
+			break;
+		case USBD_EVT_POWER:
+			usbd_work_process_pwr_events(&ev->evt.pwr_evt);
+			break;
+		case USBD_EVT_RESET:
+			LOG_DBG("USBD reset event.");
+			k_mutex_lock(&ctx->drv_lock, K_FOREVER);
+			eps_ctx_init();
+			k_mutex_unlock(&ctx->drv_lock);
+
+			if (ctx->status_cb) {
+				ctx->status_cb(USB_DC_RESET, NULL);
 			}
-			case EP_EVT_RECV_COMPLETE:
-				ep_ctx->cfg.cb(ep_ctx->cfg.addr,
-					       USB_DC_EP_DATA_OUT);
-				break;
-
-			case EP_EVT_WRITE_COMPLETE:
-				if (ep_ctx->cfg.type == USB_DC_EP_CONTROL) {
-					k_mutex_lock(&ctx->drv_lock, K_FOREVER);
-					nrfx_usbd_setup_clear();
-					k_mutex_unlock(&ctx->drv_lock);
-				}
-				ep_ctx->cfg.cb(ep_ctx->cfg.addr,
-					       USB_DC_EP_DATA_IN);
-				break;
-			default:
-				break;
+			break;
+		case USBD_EVT_SOF:
+			if (ctx->status_cb) {
+				ctx->status_cb(USB_DC_SOF, NULL);
 			}
+			break;
+		default:
+			LOG_ERR("Unknown USBD event: %d.", (u32_t)ev->evt_type);
```

Comment:
please do not cast to `u32_t`
First default argument type is int so you type will be cast to it anyway, second if specific width is used you should use appropriate formatting modifier (in this case `PRIu32`)